src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -2335,20 +2340,94 @@ public sealed override void Filter(
                     IValueSet? whenFalseValues,
                     out Tests whenTrue,
                     out Tests whenFalse,
-                    ref bool foundExplicitNullTest)
+                    ref bool foundExplicitNullTest,
+                    BindingDiagnosticBag diagnostics)
                 {
                     var trueBuilder = ArrayBuilder<Tests>.GetInstance(RemainingTests.Length);
                     var falseBuilder = ArrayBuilder<Tests>.GetInstance(RemainingTests.Length);
-                    foreach (var other in RemainingTests)
+                    for (int i = 0; i < RemainingTests.Length; i++)
                     {
-                        other.Filter(builder, test, state, whenTrueValues, whenFalseValues, out Tests oneTrue, out Tests oneFalse, ref foundExplicitNullTest);
+                        var other = RemainingTests[i];
+                        other.Filter(builder, test, state, whenTrueValues, whenFalseValues, out Tests oneTrue, out Tests oneFalse, ref foundExplicitNullTest, diagnostics);
                         trueBuilder.Add(oneTrue);
                         falseBuilder.Add(oneFalse);
+
+                        if (i > 0
+                            && oneTrue is False
```

Comment:
>&& oneTrue is False [](http://example.com/codeflow?start=28&length=19)

Could you explain and add a comment about what conclusion we draw from this fact?